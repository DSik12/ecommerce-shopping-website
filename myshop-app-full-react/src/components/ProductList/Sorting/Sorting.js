import React from 'react';
import data from '../productListConfig/productList.json';
import { lowToHigh, highToLow, bestSeller, featured } from '../productListConfig/productListStatic';

const Sorting = (props) => {

  let productApi = "";

  const checkExistingQuery = (query) => {
    const search = window.location.search;
    if (search) {
      const existingQuery = search.substring(1, search.length);
      if (existingQuery.includes(data.featured)) {
        query = existingQuery.replace(data.featured, query);
        return query;
      }
      else if (existingQuery.includes(data.sortAsc)) {
        query = existingQuery.replace(data.sortAsc, query);
        return query;
      }
      else if (existingQuery.includes(data.sortDesc)) {
        query = existingQuery.replace(data.sortDesc, query);
        return query;
      }
      else if (existingQuery.includes(data.bestSeller)) {
        query = existingQuery.replace(data.bestSeller, query);
        return query;
      }
      else {
        query = `${existingQuery}&${query}`;
        return query;
      }
    }
    else {
      return query;
    }
  };

  const SortingDropDown = (event) => {
    event.preventDefault();
    const selectedDiscount = event;
    selectedDiscount.checked = false;
    const selectedOption = document.getElementById("selectTag");
    switch (selectedOption.value) {
      case lowToHigh:
        productApi = checkExistingQuery(data.sortAsc);
        props.navigate(data.products, productApi);
        break;
      case highToLow:
        productApi = checkExistingQuery(data.sortDesc);
        props.navigate(data.products, productApi);
        break;
      case featured:
        productApi = checkExistingQuery(data.featured);
        props.navigate(data.products, productApi);
        break;
      case bestSeller:
        productApi = checkExistingQuery(data.bestSeller);
        props.navigate(data.products, productApi);
        break;
      default:

        break;
    }
  };


  return (
    <>
        <h3 className="products-heading">Products</h3>
        <div className="products140">
          <p className="products-heading-140" id="productsCount"> </p>
          <p className="filterText">Filter</p>
          <div className="forForm">

            <div className="sort">
              <span>
                <label htmlFor="filter" className="sortLabel">
                  <b>Sort &nbsp;&nbsp;</b>
                </label>
              </span>
              <span>
                <select name="filter" className="filtered" id="selectTag" onChange={SortingDropDown} data-testid="sortingTypes">
                  <option value="none" >Latest</option>
                  <option value="bestSeller">Best Sellers</option>
                  <option value="featured"> Featured Products</option>
                  <option value="lowToHigh">Price - Low to High</option>
                  <option value="highToLow">Price - High to Low</option>
                </select>
              </span>
            </div>

          </div>
        </div>
        

    </>
  )
}

export default Sorting;
