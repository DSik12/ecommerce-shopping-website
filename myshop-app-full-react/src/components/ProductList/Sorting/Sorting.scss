@import "../../shared/ScssBreakpoint/Breakpoint.scss";
// header starts/
$none: none;
.forForm {
	font-size: 10px;
	width: 33%;
	text-align: center;
	.filtered {
		font-size: 9px;
		width: 60px;
	}
}

.products140 {
	margin: 10px;
	display: flex;
	.filterText {
		font-size: 12px;
		width: 27%;
		text-align: center;
	}
	.products-heading-140 {
		font-size: 11px;
		width: 38%;
		padding-left: 2%;
	}
}

@media (min-width: $tablet-portrait) {
	.forForm .filtered {
		font-size: 11px;
		width: 80px;
	}

	.forForm {
		font-size: 20px;
		width: 50%;
		text-align: -webkit-right;
		margin-right: 8%;
	}
	.products140 {
		.filterText {
			display: $none;
		}
		.products-heading-140 {
			width: 50%;
			padding-left: 3%;
			font-size: 15px;
		}
		.sort {
			width: 150px;
		}
	}
}

@media (min-width: $laptop) {
	.forForm {
		font-size: 10px;
		padding-left: 33%;
	}
	.products140 .filterText {
		font-size: 20px;
		margin: 2% 5% 2% 5%;
	}
}
