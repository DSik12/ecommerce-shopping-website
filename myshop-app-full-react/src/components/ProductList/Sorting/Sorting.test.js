import React from 'react';
import Sorting from './Sorting';
import ReactDOM from 'react-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer'


describe("Sorting Component", () => {
    
    it('renders the comp properly', () => {
        const divEl = document.createElement('div');
        ReactDOM.render(<Sorting />, divEl);

    });

    it('should have filter as text', () => {
        render(<Sorting />);
        const filterText = screen.getByText('Filter');
        expect(filterText).toBeInTheDocument();
    });

    it('should have sort as text', () => {
        render(<Sorting />);
        const sortText = screen.getByText('Sort');
        expect(sortText).toBeInTheDocument();
    });

    it('should update on onChange event in sorting event', () => {
        render(<Sorting />);
        //mock fire the event
        const sortingTypes = screen.getByTestId('sortingTypes');
        fireEvent.change(sortingTypes, { target: { value: "" } });
        expect(sortingTypes.value).toBe("");
    })

    it('should have component snapshot', () => {
        const tree = renderer.create(<Sorting/>);
        expect(tree).toMatchSnapshot();
    })


})