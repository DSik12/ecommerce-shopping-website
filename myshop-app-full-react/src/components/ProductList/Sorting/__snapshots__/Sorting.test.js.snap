// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting Component should have component snapshot 1`] = `
Array [
  <h3
    className="products-heading"
  >
    Products
  </h3>,
  <div
    className="products140"
  >
    <p
      className="products-heading-140"
      id="productsCount"
    >
       
    </p>
    <p
      className="filterText"
    >
      Filter
    </p>
    <div
      className="forForm"
    >
      <div
        className="sort"
      >
        <span>
          <label
            className="sortLabel"
            htmlFor="filter"
          >
            <b>
              Sort   
            </b>
          </label>
        </span>
        <span>
          <select
            className="filtered"
            data-testid="sortingTypes"
            id="selectTag"
            name="filter"
            onChange={[Function]}
          >
            <option
              value="none"
            >
              Latest
            </option>
            <option
              value="bestSeller"
            >
              Best Sellers
            </option>
            <option
              value="featured"
            >
               Featured Products
            </option>
            <option
              value="lowToHigh"
            >
              Price - Low to High
            </option>
            <option
              value="highToLow"
            >
              Price - High to Low
            </option>
          </select>
        </span>
      </div>
    </div>
  </div>,
]
`;
